File: libs/workflow-core/src/models/execution.ts
export type ExecutionStatus = 'pending' | 'running' | 'completed' | 'failed' | 'waiting';

export interface WorkflowExecution {
  id: string;
  tenantId: string;
  definitionId: string;
  parentExecutionId?: string;
  dbosWorkflowId?: string;
  status: ExecutionStatus;
  currentNodeId?: string;
  context: Record<string, any>;
  params: Record<string, any>;
  error?: string;
  startedAt?: Date;
  completedAt?: Date;
  createdAt: Date;
}

export interface WorkflowResult {
  executionId: string;
  context: Record<string, any>;
  status: 'completed' | 'failed';
  error?: string;
}

File: apps/controlplane/internal/model/execution.go
package model

import "time"

type ExecutionStatus string

const (
	ExecutionPending   ExecutionStatus = "pending"
	ExecutionRunning   ExecutionStatus = "running"
	ExecutionCompleted ExecutionStatus = "completed"
	ExecutionFailed    ExecutionStatus = "failed"
	ExecutionWaiting   ExecutionStatus = "waiting"
)

type WorkflowExecution struct {
	ID                string
	TenantID          string
	DefinitionID      string
	ParentExecutionID string
	DBOSWorkflowID   string
	Status            ExecutionStatus
	CurrentNodeID     string
	Context           map[string]any
	Params            map[string]any
	Error             string
	StartedAt         *time.Time
	CompletedAt       *time.Time
	CreatedAt         time.Time
}
